pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -154,6 +154,14 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *pav1alpha1.PodAutosc
 	if err != nil {
 		return fmt.Errorf("error checking pods for revision %s: %w", pa.Labels[serving.RevisionLabelKey], err)
 	}
+	// Manually set the desired scale to 0 if initial scale is 0, because otherwise
+	// in computeActiveCondition, upon initial deploy we will miss the pc.want == 0
+	// case and PA will go from inactive to activating.
+	// Can't use pa.Status.IsScaleTargetInitialized() here because it will only be false once
```

Comment:
This seems like it means this happens on every scale-from-zero, not just the first? The only thing I can see here that stops this happening on every scale-from-zero is `want = -1`, but I think that could also be the case on an autoscaler restart (where we also have no metrics, but where  I think we _wouldnt_ want the InitialScale behaviour?)

Reply:
i tested this locally and we don't get `want = -1` once we've scaled up from zero (after we scale down, PA's desired scale is 0), so this will only happen for the first scale-from-zero

Reply:
what about after autoscaler is restarted?

Reply:
Yes, you need to restart the autoscaler to get -1 again